app/src/main/java/cm/aptoide/pt/search/SearchManager.java

```diff
@@ -59,76 +51,52 @@ public SearchManager(SharedPreferences sharedPreferences, TokenInvalidator token
-      int offset) {
-    return searchAppInStores(query, onlyTrustedApps, offset, false);
-  }
-
-  public Single<SearchResult> searchInFollowedStores(String query, boolean onlyTrustedApps,
-      int offset) {
-    return searchAppInStores(query, onlyTrustedApps, offset, true);
-  }
-
-  private Single<SearchResult> searchAppInStores(String query, boolean onlyTrustedApps, int offset,
-      boolean onlyFollowedStores) {
-    return accountManager.enabled()
-        .first()
-        .flatMap(
-            enabled -> ListSearchAppsRequest.of(query, offset, onlyFollowedStores, onlyTrustedApps,
-                StoreUtils.getSubscribedStoresIds(storeRepository), bodyInterceptor, httpClient,
-                converterFactory, tokenInvalidator, sharedPreferences, enabled,
-                appBundlesVisibilityManager)
-                .observe(false))
-        .flatMap(results -> handleSearchResults(results))
-        .onErrorResumeNext(throwable -> handleSearchError(throwable))
-        .toSingle();
-  }
-
-  public Single<SearchResult> searchInStore(String query, String storeName, int offset) {
-    return ListSearchAppsRequest.of(query, storeName, offset, subscribedStoresAuthMap,
-        bodyInterceptor, httpClient, converterFactory, tokenInvalidator, sharedPreferences,
-        appBundlesVisibilityManager)
-        .observe(false)
-        .flatMap(results -> handleSearchResults(results))
-        .onErrorResumeNext(throwable -> handleSearchError(throwable))
-        .doOnError(throwable -> throwable.printStackTrace())
-        .toSingle();
-  }
-
-  private Observable<SearchResult> handleSearchResults(ListSearchApps results) {
-    return Observable.just(results)
-        .filter(listSearchApps -> hasResults(listSearchApps))
-        .map(data -> data.getDataList()
-            .getList())
-        .flatMapIterable(list -> list)
-        .map(searchApp -> new SearchAppResult(searchApp))
-        .toList()
+  public Single<SearchResult> searchAppInStores(String query, List<Filter> filters) {
+    return accountManager.hasMatureContentEnabled()
         .first()
-        .map(list -> new SearchResult.Success(list));
-  }
-
-  private Observable<SearchResult> handleSearchError(Throwable throwable) {
-    if (throwable instanceof UnknownHostException
-        || throwable instanceof NoNetworkConnectionException) {
-      return Observable.just(new SearchResult.Error(SearchResultError.NO_NETWORK));
+        .toSingle()
+        .flatMap(matureEnabled -> searchRepository.generalSearch(query, getSearchFilters(filters),
+            matureEnabled));
+  }
+
+  public SearchFilters getSearchFilters(List<Filter> viewFilters) {
+    boolean onlyFollowedStores = false;
+    boolean onlyTrustedApps = false;
+    boolean onlyBetaApps = false;
+    boolean onlyAppcApps = false;
+    for (Filter filter : viewFilters) {
+      if (filter.getIdentifier() == null) continue;
```

Comment:
In this case no, I simply made it nullable because technically the identifier is not needed to have a fully functional view of filters (e.g. if for example every filter were to function the same way). I don't have a particular scenario for this though so I could remove it